extension/source/Confirm/Confirm.tsx

```diff
@@ -12,37 +12,37 @@ import {
 } from '../types/Rpc';
 import TransactionCard from './TransactionCard';
 import onAction from '../helpers/onAction';
+import { useQuill } from '../QuillContext';
+import useCell from '../cells/useCell';
+import Loading from '../components/Loading';
 
 const Confirm: FunctionComponent = () => {
-  const [id, setId] = useState<string | null>();
-  const [actions, setActions] = useState<SendTransactionParams[]>([]);
+  const quill = useQuill();
+
+  const id = new URL(window.location.href).searchParams.get('id');
+  const transactions = useCell(quill.cells.transactions);
+
   const [current, setCurrent] = useState<number>(0);
 
-  useEffect(() => {
-    const params = new URL(window.location.href).searchParams;
-    const txid = params.get('id');
-    setId(txid);
+  if (transactions === undefined) {
+    return <Loading />;
+  }
 
-    const fetchTx = async () => {
-      const data = await storage.local.get('transactions');
-      const tx = data.transactions.value.outgoing.find(
-        (t: any) => t.id === txid,
```

Comment:
Use of `any` here is what really prompted this. Trying to clean up all our linting issues.